hilti/runtime/include/fiber.h

```diff
@@ -48,6 +45,140 @@ using Handle = detail::Fiber;
 
 namespace detail {
 
+namespace lambda {
```

Comment:
It'd be worth moving this out of fiber into a general utility class. I can see using this at other places, too, where we currently use std::function to capture a lambda.

Reply:
I'll move it into a separate file in the hilti runtime library.